### ISSUE TITLE
[GlobalStep - WooCommerce 4.9 Beta 1] Order status is displayed as "On Hold" on dashboard for PayPal payment method .

### ISSUE THREAD

**Issue by gglobalstep** [woocommerce/woocommerce#28670]
## Bug Description:
Order status is displayed as "On Hold" on dashboard for PayPal payment method (Test Mode) on test site.
 
## Environment:
Windows 10, macOS Mojave v10.14.6 (18G103)
Browser: Chrome(Version 87.0.4280.88 (64-bit))
                Firefox(Version 83.0)(64-bit)
                Safari: v13.1.1 (14609.2.9.1.3)
 
## Steps To Reproduce:
1. Hit any one of the existing test site URL (PayPal payment method is enabled)
1. Login with valid credentials
1. Go to Frontend, place an order and do the payment using PayPal payment test account.
1. Once Payment via PayPal is completed, Go back to dashboard
1. Go to Orders.
1. Observe that Order status is displayed as "On Hold" on dashboard for PayPal payment method .

## Actual Result:
Order status is displayed as "On Hold" on dashboard for PayPal payment method .
 
## Expected Result:
Order status should be displayed as "Processing" on dashboard for PayPal payment method .

## Artifacts:
[#28670.zip](https://github.com/woocommerce/woocommerce/files/5740014/28670.zip)

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

<details>
```
'
### WordPress Environment ###

WordPress address (URL): https://lw.testwc.com
Site address (URL): https://lw.testwc.com
WC Version: 4.9.0
REST API Version: ✔ 4.9.0
WC Blocks Version: ✔ 4.0.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.8.1
Log Directory Writable: ✔
WP Version: ❌ 5.5.3 - There is a newer version of WordPress available (5.6.1-alpha-49877)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.14.0
PHP Version: 7.2.32-1+ubuntu18.04.1+deb.sury.org+1
PHP Post Max Size: 200 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 7.58.0
OpenSSL/1.1.1g

SUHOSIN Installed: –
MySQL Version: 5.6.49-89.0
Max Upload Size: 200 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.9.0
WC Database Prefix: wp_
Total Database Size: 18.14MB
Database Data Size: 15.25MB
Database Index Size: 2.89MB
wp_woocommerce_sessions: Data: 5.52MB + Index: 0.27MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.11MB + Index: 0.14MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.36MB + Index: 0.30MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.20MB + Index: 0.16MB + Engine InnoDB
wp_affiliate_wp_affiliatemeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_affiliate_wp_affiliates: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_campaigns: Data: 0.00MB + Index: 0.00MB + Engine
wp_affiliate_wp_creatives: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_customermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_affiliate_wp_customers: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_affiliate_wp_direct_links: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_affiliate_wp_payouts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_referrals: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_rest_consumers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_visits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.09MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 3.13MB + Index: 0.09MB + Engine InnoDB
wp_postmeta: Data: 1.48MB + Index: 0.45MB + Engine InnoDB
wp_posts: Data: 3.38MB + Index: 0.13MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wcs_payment_retries: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpmerge_options: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 85
customize_changeset: 1
log_emails_log: 132
nav_menu_item: 1
page: 12
post: 24
product: 53
product_variation: 12
revision: 125
shop_coupon: 4
shop_order: 69
shop_order_refund: 1
shop_subscription: 6

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (19) ###

Akismet Anti-Spam: by Automattic – 4.1.7
Astra Pro: by Brainstorm Force – 2.5.1
Classic Editor: by WordPress Contributors – 1.6
Facebook for WooCommerce: by Facebook – 2.2.0
Gutenberg: by Gutenberg Team – 9.5.2
Jetpack by WordPress.com: by Automattic – 9.2.1
Google Ads & Marketing by Kliken: by Kliken – 1.0.6
Log Emails: by WebAware – 1.4.0
Mailchimp for WooCommerce: by Mailchimp – 2.5.0
Password Protected: by Ben Huson – 2.4
Compress JPEG & PNG images: by TinyPNG – 3.2.1
Razorpay for WooCommerce: by Team Razorpay – 2.5.0
WooCommerce Beta Tester: by WooCommerce – 2.0.2
WooCommerce PayPal Checkout Gateway: by WooCommerce – 2.1.1
WooCommerce Stripe Gateway: by WooCommerce – 4.7.0
WooCommerce Shipping & Tax: by Automattic – 1.25.4
WooCommerce Subscriptions: by WooCommerce – 3.0.10
WooCommerce: by Automattic – 4.9.0-beta.1
WordPress Beta Tester: by Peter Westwood
Andy Fragen – 3.0.9


### Inactive Plugins (0) ###


### Must Use Plugins (4) ###

Liquid Web Managed WordPress: by Liquid Web <support@liquidweb.com> – 1.4
Liquid Web Managed WordPress Configuration: by Liquid Web – 1.0.0
Liquid Web Managed WordPress Platform Manager: by Liquid Web – 1.1.0
Liquid Web Update Manager: by Liquid Web <support@liquidweb.com> – 1.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #738 - /shop/
Cart: #739 - /cart/
Checkout: #740 - /checkout/
My account: #741 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-One
Version: 1.1
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://lw.testwc.com
Subscription Statuses: wc-active: 5
wc-expired: 1

WooCommerce Account Connected: ✔ Yes
Active Product Key: ✔ Yes
Custom Retry Rules: ✔ No
Custom Retry Rule Class: ✔ No
Custom Raw Retry Rule: ✔ No
Custom Retry Rule: ✔ No
Retries Migration Status: ✔ Completed
Report Cache Enabled: ✔ Yes
Cache Update Failures: ✔ 0 failure

### Store Setup ###

Country / State: India — Maharashtra

### Subscriptions by Payment Gateway ###

Stripe: wc-active: 5
wc-expired: 1


### Payment Gateway Support ###

Cash on delivery: products
PayPal Standard: subscription_payment_method_change_customer
subscription_payment_method_change_admin
subscription_amount_changes
subscription_date_changes
multiple_subscriptions
subscription_payment_method_delayed_change
subscriptions
subscription_cancellation
subscription_suspension
subscription_reactivation
products
refunds

PayPal Reference Transactions Enabled: ✔ Yes
Stripe: products
refunds
tokenization
add_payment_method
subscriptions
subscription_cancellation
subscription_suspension
subscription_reactivation
subscription_amount_changes
subscription_date_changes
subscription_payment_method_change
subscription_payment_method_change_customer
subscription_payment_method_change_admin
multiple_subscriptions
pre-orders

Razorpay: products
refunds

PayPal Checkout: products
refunds
subscriptions
subscription_cancellation
subscription_reactivation
subscription_suspension
multiple_subscriptions
subscription_payment_method_change_customer
subscription_payment_method_change_admin
subscription_amount_changes
subscription_date_changes


### Action Scheduler ###

Complete: 846
Oldest: 2020-11-23 14:49:18 +0000
Newest: 2020-12-24 13:49:23 +0000

Pending: 5
Oldest: 2020-12-24 14:49:23 +0000
Newest: 2021-01-19 21:52:25 +0000

`
```
</details>

**Comment by juliaamosova** [comment]
Hi @gglobalstep,

Thank you for reporting the issue!

I've spent some understanding of what could be happening. I am getting the following order notes on the PayPal Sandbox orders:

![pp](https://user-images.githubusercontent.com/19143190/103309881-468dc780-49e4-11eb-9ea2-bad0408e408f.jpg)

There are numerous threads available online on possible solutions for this issue. The closest one I found so far is this one:

https://stackoverflow.com/questions/10171394/paypal-ipn-returns-pending-reason-unilateral

It seems there is a mismatch of the emails used between PayPal sandbox mode (used on the site to set up PayPal) and the test PayPal customer email (provided by PayPal). In short, it looks like a misconfiguration. I will dive into this a bit further later to understand a proper config.
